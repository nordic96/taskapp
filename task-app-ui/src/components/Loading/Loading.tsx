import React from 'react';
import { Box } from '@mui/material';
import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css'

const LoadingBox = () => {
    return (
        <Box width={'40vw'} display={'flex'} flexDirection={'column'} justifyContent={'flex-start'}>
            <Skeleton width={'100%'} height={90} />
            <Skeleton width={'95%'} height={10} />
        </Box>
    );
};

const Loading = () => {
    return (
        <Box id={'loading-container'} sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
            {new Array(4).fill(0).map((x, i) => {
                return <LoadingBox key={i} />;
            })}
        </Box>
    );
};

export default Loading;
