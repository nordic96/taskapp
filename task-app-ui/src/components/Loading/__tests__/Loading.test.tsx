import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../Loading';

describe('Testing Loading Component', () => {
    it('Renders and matches snapshot', () => {
        const wrapper = shallow(<Loading />);
        expect(wrapper).toMatchSnapshot();
    });
});
