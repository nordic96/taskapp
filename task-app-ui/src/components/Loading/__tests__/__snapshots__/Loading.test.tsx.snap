// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Component Renders and matches snapshot 1`] = `ShallowWrapper {}`;
